### PR TITLE
Add debug

### PR DESCRIPTION
**Pull request by n1tram1**

### CODE REVIEW

File: module/mfrc522_module.c [hunk 1]
```diff
@@ -17,20 +18,76 @@
...
+		break;
+	case MFRC522_CMD_MEM_WRITE:
+		do_debug_write(cmd->data);
+		break;
+	default:
+		/* Do nothing */
+		break;
+	}
+}
+
```
Comment (Skallwar) [inline module/mfrc522_module.c:90]: Move this to mfrc522_debug.c

File: module/mfrc522_module.c [hunk 2]
```diff
@@ -48,14 +105,17 @@ static ssize_t __mfrc522_write(struct mfrc522_state *state, const char *buffer,
 	if (command.data[0])
 		pr_info("[MFRC522] With extra data: `%s`\n", command.data);
 
-	answer_size = mfrc522_execute(state->answer, &command);
+	answer_size = mfrc522_execute(state, state->answer, &command);
```
Comment (Skallwar) [inline module/mfrc522_module.c:108]: We could stop to use both ``state`` and ``state->answer``

**Review by Skallwar — CHANGES_REQUESTED** [review]

File: module/mfrc522_module.c [hunk 3]
```diff
@@ -17,20 +18,76 @@
...
+	for (i = 1; i < answer_size + 1; i++) {
+		write_head +=
+			snprintf(line + write_head, 16, "%02x", answer[i - 1]);
+
+		if (i % 5 == 0) {
+			pr_info("%s\n", line);
+			write_head = 0;
+		} else {
+			write_head += snprintf(line + write_head, 16, " ");
+		}
```
Comment (CohenArthur) [inline module/mfrc522_module.c:44]: We could put this part in a separate function to avoir repetition

File: module/mfrc522_module.c [hunk 4]
```diff
@@ -17,20 +18,76 @@
...
+	for (i = 1; cmd[i - 1]; i++) {
+		write_head +=
+			snprintf(line + write_head, 16, "%02x", cmd[i - 1]);
+
+		if (i % 5 == 0) {
+			pr_info("%s\n", line);
+			write_head = 0;
+		} else {
+			write_head += snprintf(line + write_head, 16, " ");
+		}
```
Comment (CohenArthur) [inline module/mfrc522_module.c:68]: Same as above

File: module/mfrc522_module.c [hunk 5]
```diff
@@ -198,6 +258,8 @@ static int __init mfrc522_init(void)
 		.fops = &mfrc522_fops,
 	};
 
+	state->debug_on = false;
```
Comment (CohenArthur) [inline module/mfrc522_module.c:261]: We disable debugging after each write?

**Pull request merged**